wgpu/gui/base.py

```diff
@@ -247,11 +247,12 @@ def handle_event(self, event):
 
     def _dispatch_pending_events(self):
         """Handle any pending rate-limited events."""
-        events = self._pending_events.values()
-        self._last_event_time = time.perf_counter()
-        self._pending_events = {}
-        for ev in events:
-            self._dispatch_event(ev)
+        if self._pending_events:
+            events = self._pending_events.values()
+            self._last_event_time = time.perf_counter()
+            self._pending_events = {}
+            for ev in events:
+                self.handle_event(ev)
```

Comment:
Two changes really:
* Wrap this up in a `if self._pending_events:`
* Call `self.handle_event(ev)` instead of `self._dispatch_event(ev)`